File: templates/Uwp/_comp/_shared/Service.IdentityCommon._VB/Param_ProjectName/Helpers/ImageHelper$wts.ItemName_gpostaction.vb
'{[{
Imports System.Runtime.InteropServices.WindowsRuntime
Imports Windows.Storage.Streams
'}]}
Namespace Helpers
    Module ImageHelper
'^^
'{[{
        Async Function ImageFromStringAsync(data As String) As Task(Of BitmapImage)
            Dim byteArray = Convert.FromBase64String(data)
            Dim image = New BitmapImage()

            Using stream = New InMemoryRandomAccessStream()
                Await stream.WriteAsync(byteArray.AsBuffer())
                stream.Seek(0)
                Await image.SetSourceAsync(stream)
            End Using

            Return image
        End Function

        Function ImageFromAssetsFile(fileName As String) As BitmapImage
            Dim image = New BitmapImage(New Uri($"ms-appx:///Assets/{fileName}"))
            Return image
        End Function
'}]}
    End Module
End Namespace

File: templates/Uwp/_comp/_shared/Service.IdentityCommon/Param_ProjectName/Helpers/ImageHelper$wts.ItemName_gpostaction.cs
//{[{
using System.Runtime.InteropServices.WindowsRuntime;
using Windows.Storage.Streams;
//}]}

namespace Param_RootNamespace.Helpers
{
    public static class ImageHelper
    {
//^^
//{[{

        public static async Task<BitmapImage> ImageFromStringAsync(string data)
        {
            var byteArray = Convert.FromBase64String(data);
            var image = new BitmapImage();
            using (var stream = new InMemoryRandomAccessStream())
            {
                await stream.WriteAsync(byteArray.AsBuffer());
                stream.Seek(0);
                await image.SetSourceAsync(stream);
            }

            return image;
        }

        public static BitmapImage ImageFromAssetsFile(string fileName)
        {
            var image = new BitmapImage(new Uri($"ms-appx:///Assets/{fileName}"));
            return image;
        }
//}]}
    }
}
